File: 2478-longest-nice-subarray/2478-longest-nice-subarray.py
class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        maxcount = 1
        
        for i in range(len(nums)):
            temp = nums[i]
            count = 1
            
            for j in range(i + 1, len(nums)):
                if (temp & nums[j]) == 0:  
                    temp += nums[j]
                    count += 1
                else:
                    break
            
            maxcount = max(maxcount, count)
        
        return maxcount

File: 2478-longest-nice-subarray/2478-longest-nice-subarray.java
class Solution {
    public int longestNiceSubarray(int[] nums) {
        int maxcount = 1;
        for (int i = 0; i < nums.length; i++) { 
            int temp = nums[i];
            int count = 1;
            for (int j = i + 1; j < nums.length; j++) {
                if ((temp & nums[j]) == 0) { 
                    temp |= nums[j];
                    count++;
                } else {
                    break;
                }
            }
            maxcount = Math.max(maxcount, count);
        }
        return maxcount;
    }
}
